app/controllers/submissions_controller.rb

```diff
@@ -0,0 +1,32 @@
+class SubmissionsController < ApplicationController
+  def create
+    problem = Submission.new(create_params)
+    if problem.save!
+      render status: :created, json: problem
+    end
+  end
+
+  def show
+    problem = Submission.find(params[:id])
+    render status: :ok, json: problem
+  end
+
+  def update
+    problem = Submission.find(params[:id])
+    if problem.update!(create_params)
+      render status: :ok, json: problem
+    end
+  end
+
+  def destroy
+    problem = Submission.find(params[:id])
+    problem.destroy
+    head :no_content
+  end
+
+  private
+
+  def create_params
+    params.permit(:name, :description, :assignment_id, :filepath)
```

Comment:
Updated this to accurately reflect creation params for Submissions. `permit` acts as a filter for parameters, see https://apidock.com/rails/ActionController/Parameters/permit.